compiler/parser/parser.cpp

```diff
@@ -871,6 +930,17 @@ static ModuleSymbol* parseFile(const char* path,
   return retval;
 }
 
+// TODO: Add helpers to convert locations without passing IDs.
+static void uASTParseFileErrorHandler(const chpl::ErrorMessage& err) {
+  auto markError = astlocMarker(err.location());
```

Comment:
Are you using `auto` over `astlocMarker markError(err.location())` for a specific purpose?

Reply:
No, I just prefer to write it that way, but that might be C++'tastic so I'll defer to your judgement. I was worried at one point that it might introduce a copy but I tested with MSVC (perhaps not the best metric lol) and it didn't seem like that was the case?

Reply:
I also tend to like this auto version. I've seen some C++ talks about how the auto version reduces a syntactic confusion / adds consistency because you can use the same pattern with a zero-args constructor e.g. `auto mark = astLocMarker();` where `astlocMarker mark();` is an error.

Reply:
(FWIW it is also more similar to Chapel forms which people working on our compiler are probably familiar with).

Reply:
Apart from that though I'm not seeing why `astlocMarker` is used here. I would expect instead that the Location is passed as the first argument to `USR_FATAL_CONT` / `USR_PRINT` etc. The `astlocMarker` is updating a global sortof last resort location that is used when constructing AST nodes most of all.

Reply:
I just followed the pattern used in the dispatcher in `convert-uast.cpp`, but I can look into passing the location to `USR_*` stuff as a followup?